base/src/org/compiere/process/OrderPOCreate.java

```diff
@@ -38,54 +44,50 @@
  *  Contributor: Carlos Ruiz - globalqss
  *      Fix [1709952] - Process: "Generate PO from Sales order" bug
  */
-public class OrderPOCreate extends SvrProcess
+public class OrderPOCreate extends OrderPOCreateAbstract
 {
-	/**	Order Date From		*/
-	private Timestamp	p_DateOrdered_From;
-	/**	Order Date To		*/
-	private Timestamp	p_DateOrdered_To;
-	/**	Customer			*/
-	private int			p_C_BPartner_ID;
-	/**	Vendor				*/
-	private int			p_Vendor_ID;
-	/**	Sales Order			*/
-	private int			p_C_Order_ID;
-	/** Drop Ship			*/
-	private boolean		p_IsDropShip = false;
 	
+	private int m_explosion_level = 0;
+	private Set<MPPProductBOMLine> bomSet = new HashSet<MPPProductBOMLine>(); 
+	private int explosion(MPPProductBOM bom) throws AdempiereUserError {

```

Comment:
Hi @homebeaver i don't know what is the reason for add BOM process here, BOM process is created when you complete a Sales Order, see it here: https://github.com/adempiere/adempiere/blob/develop/base/src/org/compiere/model/MOrder.java#L1239

All changes is ok but in don't agree with add a separed process for make **exprosion** is it is part of business model for MOrder

Best regard

Reply:
@yamelsenih , let me explain:

if you have a BOM-product in your sales order and you create the purchase order from it, so you cannot purchase the BOM-product itsself, You have to purchase all the components and produce the product (make-to-order). This was not considered in `OrderPOCreate` .
I did not implement purchase components, nor make-to-order. This is heavy work. 
But at least I send a warning:
```
   // TODO purchase component and prepare BOM production
  log.warning(mProduct.toString() + " is BOM product, will not be purchased (not implemented)");
```
It's up to you (or us) to decide what is better
- send a warning on not implemeted feature
- do nothing , remove the explosion and do not inform the user about the hole

This have some interfaces to production.
What do you think? What is the opinion of Mario @marcalwestf  ?
regards

Reply:
see my answer below